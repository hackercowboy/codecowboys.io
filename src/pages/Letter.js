import React from 'react';
import PropTypes from 'prop-types';
import { Parallax } from 'react-parallax';
import { injectIntl } from 'react-intl';

import ApplicationCarousel from '../components/ApplicationCarousel';
import ApplicationHero from '../components/ApplicationHero';
import ContactForm from '../components/ContactForm';
import Feature from '../components/Feature';
import FeatureGroup from '../components/FeatureGroup';
import FeatureImage from '../components/FeatureImage';
import Features from '../components/Features';
import Header from '../components/Header';
import Main from '../components/Main';
import Navigation from '../components/Navigation';
import NavigationItem from '../components/NavigationItem';
import Screenshots from '../components/Screenshots';
import ScrollButton from '../components/ScrollButton';
import Section from '../components/Section';

const letterCarouselImage = require('../assets/images/letter_carousel.webp');
const heroBackground = require('../assets/images/bg-7.webp');
const letterFeatures = require('../assets/images/letter_features.webp');

function Letter({ intl }) {
  const ipadImages = [1, 2, 3, 4, 5].map((index) => ({
    original: `/screenshots/letter/letter_ipad_${index}_de.webp`,
    thumbnail: `/screenshots/letter/letter_ipad_${index}_de.webp`,
  }));

  const iphoneImages = [1, 2, 3, 4, 5].map((index) => ({
    original: `/screenshots/letter/letter_iphone_${index}_de.webp`,
    thumbnail: `/screenshots/letter/letter_iphone_${index}_de.webp`,
  }));

  return (
    <Main className="presentation">
      <Header>
        <Navigation>
          <NavigationItem id="navigation-item-features" target="features" offset={-94}>{intl.formatMessage({ id: 'letter.features' })}</NavigationItem>
          <NavigationItem id="navigation-item-description" target="description" offset={-88}>{intl.formatMessage({ id: 'letter.description' })}</NavigationItem>
          <NavigationItem id="navigation-item-screenshots" target="screenshots" offset={-94}>{intl.formatMessage({ id: 'letter.screenshots' })}</NavigationItem>
          <NavigationItem id="navigation-item-contact" target="contact" offset={-94}>{intl.formatMessage({ id: 'contact.title' })}</NavigationItem>
        </Navigation>
      </Header>
      <Parallax
        bgImage={heroBackground}
        strength={500}
      >
        <ApplicationCarousel>
          <ApplicationHero
            image={letterCarouselImage}
            title={intl.formatMessage({ id: 'letter.title' })}
            appStoreLink="http://itunes.apple.com/app/letter/id498506154"
            description={intl.formatMessage({ id: 'letter.teaser' })}
          />
        </ApplicationCarousel>
        <ScrollButton target="features" />
      </Parallax>
      <Section
        id="features"
        title={intl.formatMessage({ id: 'letter.features' })}
        subtitle={intl.formatMessage({ id: 'letter.features_subtitle' })}
      >
        <Features>
          <FeatureGroup>
            <Feature
              title={intl.formatMessage({ id: 'letter.features_print_title' })}
              description={intl.formatMessage({ id: 'letter.features_print_description' })}
              icon="fa-print"
            />
            <Feature
              title={intl.formatMessage({ id: 'letter.features_storage_title' })}
              description={intl.formatMessage({ id: 'letter.features_storage_description' })}
              icon="fa-cloud"
            />
            <Feature
              title={intl.formatMessage({ id: 'letter.features_search_title' })}
              description={intl.formatMessage({ id: 'letter.features_search_description' })}
              icon="fa-search"
            />
          </FeatureGroup>
          <FeatureImage
            src={letterFeatures}
            width={280}
            alt={intl.formatMessage({ id: 'letter.features' })}
          />
          <FeatureGroup>
            <Feature
              title={intl.formatMessage({ id: 'letter.features_addresses_title' })}
              description={intl.formatMessage({ id: 'letter.features_addresses_description' })}
              icon="fa-user"
              right
            />
            <Feature
              title={intl.formatMessage({ id: 'letter.features_signature_title' })}
              description={intl.formatMessage({ id: 'letter.features_signature_description' })}
              icon="fa-pencil"
              right
            />
            <Feature
              title={intl.formatMessage({ id: 'letter.features_templates_title' })}
              description={intl.formatMessage({ id: 'letter.features_templates_description' })}
              icon="fa-file-alt"
              right
            />
          </FeatureGroup>
        </Features>
      </Section>
      <Section
        id="description"
        title={intl.formatMessage({ id: 'letter.description' })}
        subtitle={intl.formatMessage({ id: 'letter.description_subtitle' })}
      >
        <p>{intl.formatMessage({ id: 'letter.description_1' })}</p>
        <p>{intl.formatMessage({ id: 'letter.description_2' })}</p>
        <p>{intl.formatMessage({ id: 'letter.description_3' })}</p>
        <p>{intl.formatMessage({ id: 'letter.description_4' })}</p>
        <p>{intl.formatMessage({ id: 'letter.description_5' })}</p>
      </Section>
      <Section
        id="screenshots"
        title={intl.formatMessage({ id: 'letter.screenshots' })}
        subtitle={intl.formatMessage({ id: 'letter.screenshots_subtitle' })}
      >
        <Screenshots ipadImages={ipadImages} iphoneImages={iphoneImages} />
      </Section>
      <Section
        id="contact"
        title={intl.formatMessage({ id: 'contact.title' })}
        subtitle={intl.formatMessage({ id: 'contact.subtitle' })}
      >
        <ContactForm />
      </Section>
    </Main>
  );
}

Letter.propTypes = {
  intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }).isRequired,
};

export default injectIntl(Letter);
