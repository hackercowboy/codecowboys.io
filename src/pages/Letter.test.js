import React from 'react';
import { shallow } from 'enzyme';

import Letter from './Letter';

describe('<Letter/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Letter />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });
});
