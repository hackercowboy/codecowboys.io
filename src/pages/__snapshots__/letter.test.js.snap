// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Letter/> should render 1`] = `
"<Page className=\\"presentation\\">
  <Header>
    <Navigation>
      <NavigationItem id=\\"navigation-item-features\\" target=\\"features\\" offset={-94}>
        Features
      </NavigationItem>
      <NavigationItem id=\\"navigation-item-description\\" target=\\"description\\" offset={-88}>
        Description
      </NavigationItem>
      <NavigationItem id=\\"navigation-item-screenshots\\" target=\\"screenshots\\" offset={-94}>
        Screenshots
      </NavigationItem>
      <NavigationItem id=\\"navigation-item-contact\\" target=\\"contact\\" offset={-94}>
        Contact
      </NavigationItem>
    </Navigation>
  </Header>
  <Parallax bgImage=\\"/images/bg-7.webp\\" strength={500} bgClassName=\\"react-parallax-bgimage\\" bgImageAlt=\\"\\" className=\\"\\" contentClassName=\\"\\" disabled={false}>
    <ApplicatinCarousel>
      <ApplicationHero image=\\"/images/letter_carousel.webp\\" title=\\"Letter\\" appStoreLink=\\"http://itunes.apple.com/app/letter/id498506154\\" description=\\"With the Letter app you can now write your personal letters on the iPhone, iPad and iPod Touch too, when and where you want.\\" soon={false} link={[undefined]} />
    </ApplicatinCarousel>
    <ScrollButton target=\\"features\\" />
  </Parallax>
  <Section id=\\"features\\" title=\\"Features\\" subtitle=\\"Overview of all features\\" dark={false}>
    <Features>
      <FeatureGroup>
        <Feature title=\\"Printing & Sharing\\" description=\\"Letters can be printed (AirPrint), sent by email, and opened in other applications as PDF or image.\\" icon=\\"fa-print\\" />
        <Feature title=\\"Storage\\" description=\\"All letters are stored securely and synchronized between devices using iCloud.\\" icon=\\"fa-cloud\\" />
        <Feature title=\\"Search\\" description=\\"With the search function letters can be found again quickly and easily.\\" icon=\\"fa-search\\" />
      </FeatureGroup>
      <FeatureImage src=\\"/images/letter_features.webp\\" width={280} alt=\\"Features\\" height={[undefined]} />
      <FeatureGroup>
        <Feature title=\\"Addresses\\" description=\\"International addresses are formatted automatically. Senders and receivers can easily be selected from your address book.\\" icon=\\"fa-user\\" right={true} />
        <Feature title=\\"Signature\\" description=\\"Letters can be signed directly in the application.\\" icon=\\"fa-pencil\\" right={true} />
        <Feature title=\\"Templates\\" description=\\"Text templates can be used to create ready letter in just a few seconds.\\" icon=\\"fa-file-alt\\" right={true} />
      </FeatureGroup>
    </Features>
  </Section>
  <Section id=\\"description\\" title=\\"Description\\" subtitle=\\"The Letter App in detail\\" dark={false}>
    <p>
      With the Letter app you can now write your personal letters on the iPhone, iPad and iPod Touch too, when and where you want.
    </p>
    <p>
      Your letters are stored and synchronized between devices using iCloud. They can be freely printed via AirPrint, sent by email and opened in other iOS applications in PDF format. The integrated search also makes finding things a breeze in many documents.
    </p>
    <p>
      The lean design is focused on the essentials to make creating the contents of your letter fast and uncomplicated. From now on tedious formatting is a thing of the past.
    </p>
    <p>
      Senders and receivers can easily be selected from your address book. International addresses are formatted automatically according to the applicable standard.
    </p>
    <p>
      With your own text templates you can create ready letters in just a few seconds.
    </p>
  </Section>
  <Section id=\\"screenshots\\" title=\\"Screenshots\\" subtitle=\\"The main screens of the Letter App\\" dark={false}>
    <Screenshots ipadImages={{...}} iphoneImages={{...}} />
  </Section>
  <Section id=\\"contact\\" title=\\"Contact\\" subtitle=\\"Please contact me if you have any questions\\" dark={false}>
    <ContactForm />
  </Section>
  <Footer />
</Page>"
`;
